resources/js/beatmapsets-show/controller.ts

```diff
@@ -70,6 +82,41 @@ export default class Controller {
     return this.beatmaps.get(this.currentBeatmap.mode) ?? [];
   }
 
+  @computed
+  get relatedTags() {
+    const map = new Map<number, TagJson>();
+
+    for (const tag of this.beatmapset.related_tags) {
+      map.set(tag.id, tag);
+    }
+
+    return map;
+  }
+
+  @computed
+  get tags() {
+    const userTags: TagJsonWithCount[] = [];
+
+    if (this.currentBeatmap.top_tag_ids != null) {
+      for (const tagId of this.currentBeatmap.top_tag_ids) {
+        const maybeTag = this.relatedTags.get(tagId.tag_id);
+        if (maybeTag == null) continue;
+
+        const tag = asTagJsonWithCount(maybeTag);
+        tag.count = tagId.count;
+        userTags.push(asTagJsonWithCount(tag));
```

Comment:
can just push `{ ...maybeTag, count: tagId.count }`

(also why is it converting the thing twice)